processor/agentmetricsprocessor/utils_calculate_utilizations.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agentmetricsprocessor
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// The following code calculates a new utilization metric from
+// a usage metric across one label (dimension) using the formula:
+//
+// value{l1=v1,...} = value{l1=v1,...} / sum(value{l1=vx,...}) over x=1..N
+
+const (
+	cpuTime         = "system.cpu.time"
+	memoryUsage     = "system.memory.usage"
+	fileSystemUsage = "system.filesystem.usage"
+	swapUsage       = "system.swap.usage"
+)
+
+var metricsToComputeUtilizationFor = map[string]bool{
+	cpuTime:         true,
+	memoryUsage:     true,
+	fileSystemUsage: true,
+	swapUsage:       true,
+}
+
+const stateLabel = "state"
+
+func (mtp *agentMetricsProcessor) appendUtilizationMetrics(rms pdata.ResourceMetricsSlice) error {
+	for i := 0; i < rms.Len(); i++ {
+		ilms := rms.At(i).InstrumentationLibraryMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			metrics := ilms.At(j).Metrics()
+			for k := 0; k < metrics.Len(); k++ {
+				metric := metrics.At(k)
+
+				// ignore all metrics except the ones we want to compute utilizations for
+				metricName := metric.MetricDescriptor().Name()
+				if _, ok := metricsToComputeUtilizationFor[metricName]; !ok {
```

Comment:
Given that you're using `map[string]bool` and not `map[string]struct{}` / set, you can simplify this to
```go
if metricsToComputeUtilizationFor[metricName] {
```